src/System.Reflection.DispatchProxy/src/System.Reflection.DispatchProxy.csproj

```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(NetFrameworkMinimum);netstandard2.0</TargetFrameworks>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <StrongNameKeyId>Open</StrongNameKeyId>
+    <IsPackable>false</IsPackable>
```

Comment:
Before merging, turn this to true and see if APICompt / PackageValidation complains about dropped TFMs.

Reply:
I got these errors, which I'm now investigating how to fix:

```
$ dotnet pack

MSBuild version 17.10.0-preview-24060-03+1725b247e for .NET
Restore complete (0.7s)
  System.Reflection.DispatchProxy net462 succeeded (1.9s) → C:\Users\calope\source\repos\maintenance-packages\artifacts\bin\System.Reflection.DispatchProxy\Release\net462\System.Reflection.DispatchProxy.dll
  System.Reflection.DispatchProxy netstandard2.0 succeeded (1.3s) → C:\Users\calope\source\repos\maintenance-packages\artifacts\bin\System.Reflection.DispatchProxy\Release\netstandard2.0\System.Reflection.DispatchProxy.dll
  System.Reflection.DispatchProxy failed with errors (0.6s)
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error PKV006: Target framework .NETStandard,Version=v1.3 is no longer supported in the latest version. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error PKV006: Target framework .NETStandard,Version=v1.3 is no longer supported in the latest version. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error PKV006: Target framework .NETCore,Version=v5.0 is no longer supported in the latest version. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error PKV006: Target framework .NETCore,Version=v5.0 is no longer supported in the latest version. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API compatibility errors between 'ref/net461/System.Reflection.DispatchProxy.dll' (C:\Users\calope\.nuget\packages\system.reflection.dispatchproxy\4.7.1\system.reflection.dispatchproxy.4.7.1.nupkg) and 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\source\repos\maintenance-packages\artifacts\packages\Release\Shipping\System.Reflection.DispatchProxy.4.8.0-dev.24069.1.nupkg): [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0003: [Baseline] ref/net461/System.Reflection.DispatchProxy.dll assembly public key token 'b03f5f7f11d50a3a' does not match with lib/netstandard2.0/System.Reflection.DispatchProxy.dll 'cc7b13ffcd2ddd51'. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API compatibility errors between 'lib/net461/System.Reflection.DispatchProxy.dll' (C:\Users\calope\.nuget\packages\system.reflection.dispatchproxy\4.7.1\system.reflection.dispatchproxy.4.7.1.nupkg) and 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\source\repos\maintenance-packages\artifacts\packages\Release\Shipping\System.Reflection.DispatchProxy.4.8.0-dev.24069.1.nupkg): [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0003: [Baseline] lib/net461/System.Reflection.DispatchProxy.dll assembly public key token 'b03f5f7f11d50a3a' does not match with lib/netstandard2.0/System.Reflection.DispatchProxy.dll 'cc7b13ffcd2ddd51'. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API compatibility errors between 'ref/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\.nuget\packages\system.reflection.dispatchproxy\4.7.1\system.reflection.dispatchproxy.4.7.1.nupkg) and 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\source\repos\maintenance-packages\artifacts\packages\Release\Shipping\System.Reflection.DispatchProxy.4.8.0-dev.24069.1.nupkg): [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0003: [Baseline] ref/netstandard2.0/System.Reflection.DispatchProxy.dll assembly public key token 'b03f5f7f11d50a3a' does not match with lib/netstandard2.0/System.Reflection.DispatchProxy.dll 'cc7b13ffcd2ddd51'. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API compatibility errors between 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\.nuget\packages\system.reflection.dispatchproxy\4.7.1\system.reflection.dispatchproxy.4.7.1.nupkg) and 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\source\repos\maintenance-packages\artifacts\packages\Release\Shipping\System.Reflection.DispatchProxy.4.8.0-dev.24069.1.nupkg): [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0003: [Baseline] lib/netstandard2.0/System.Reflection.DispatchProxy.dll assembly public key token 'b03f5f7f11d50a3a' does not match with lib/netstandard2.0/System.Reflection.DispatchProxy.dll 'cc7b13ffcd2ddd51'. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API compatibility errors between 'ref/uap10.0.16299/System.Reflection.DispatchProxy.dll' (C:\Users\calope\.nuget\packages\system.reflection.dispatchproxy\4.7.1\system.reflection.dispatchproxy.4.7.1.nupkg) and 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\source\repos\maintenance-packages\artifacts\packages\Release\Shipping\System.Reflection.DispatchProxy.4.8.0-dev.24069.1.nupkg): [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0003: [Baseline] ref/uap10.0.16299/System.Reflection.DispatchProxy.dll assembly public key token 'b03f5f7f11d50a3a' does not match with lib/netstandard2.0/System.Reflection.DispatchProxy.dll 'cc7b13ffcd2ddd51'. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API compatibility errors between 'lib/netcoreapp2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\.nuget\packages\system.reflection.dispatchproxy\4.7.1\system.reflection.dispatchproxy.4.7.1.nupkg) and 'lib/netstandard2.0/System.Reflection.DispatchProxy.dll' (C:\Users\calope\source\repos\maintenance-packages\artifacts\packages\Release\Shipping\System.Reflection.DispatchProxy.4.8.0-dev.24069.1.nupkg): [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error CP0003: [Baseline] lib/netcoreapp2.0/System.Reflection.DispatchProxy.dll assembly public key token 'b03f5f7f11d50a3a' does not match with lib/netstandard2.0/System.Reflection.DispatchProxy.dll 'cc7b13ffcd2ddd51'. [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
    C:\Users\calope\.nuget\packages\microsoft.dotnet.apicompat.task\9.0.100-alpha.1.23560.11\build\Microsoft.NET.ApiCompat.ValidatePackage.targets(39,5): error : API breaking changes found. If those are intentional, the APICompat suppression file can be updated by rebuilding with '/p:ApiCompatGenerateSuppressionFile=true' [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]

Build failed with errors in 3.7s
```

Reply:
Hold on - Did I also need to bump the versions as if I were to publish a new package? I didn't do that.

But if I bump VersionPrefix, AssemblyVersion (except the one with the comment) and PackageValidationBaseLineVersion, then I also get errors, but fewer:

```
dotnet pack
MSBuild version 17.10.0-preview-24060-03+1725b247e for .NET
    C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj : error NU1102: Unable to find package System.Reflection.DispatchProxy with version (= 4.8.0) [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj : error NU1102:   - Found 44 version(s) in dotnet-public [ Nearest version: 5.0.0-preview.1.20120.5 ] [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj : error NU1102:   - Found 1 version(s) in dotnet-eng [ Nearest version: 5.0.0-alpha.1.19618.1 ] [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\library-packs [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]
C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj : error NU1102:   - Found 0 version(s) in dotnet9 [C:\Users\calope\source\repos\maintenance-packages\src\System.Reflection.DispatchProxy\src\System.Reflection.DispatchProxy.csproj]

Restore failed with errors in 1.1s
```

Reply:
Ok nevermind, the second attempt is wrong. The first one is the correct one: IsPackable contains the _new_ version we would publish. The first time we would not change anything except IsPackable=true.

Reply:
Left a newer comment with a question: https://github.com/dotnet/maintenance-packages/pull/34#discussion_r1459842967